glide-core/src/client/value_conversion.rs

```diff
@@ -567,6 +663,14 @@ pub(crate) fn expected_type_for_cmd(cmd: &Cmd) -> Option<ExpectedReturnType> {
             }
         }
         b"LOLWUT" => Some(ExpectedReturnType::Lolwut),
+        b"GEOSEARCH" => {
+            let with_dist = cmd.position(b"WITHDIST").is_some() as u8;
+            let with_hash = cmd.position(b"WITHHASH").is_some() as u8;
+            let with_coord = cmd.position(b"WITHCOORD").is_some() as u8;
+            let flag_count = with_dist + with_hash + with_coord;
+
+            Some(ExpectedReturnType::GeoSearchReturnType(flag_count))
```

Comment:
I think you can move the check from line 404 (good number!) to here:
```suggestion
            if cmd.position(b"WITHDIST").is_some() || cmd.position(b"WITHHASH").is_some() || cmd.position(b"WITHCOORD").is_some()
                Some(ExpectedReturnType::GeoSearchReturnType)
            else
                None
```
And this releases `GeoSearchReturnType` of storing any data inside.
Sorry I didn't get this idea earlier.